src/chainparams.cpp

```diff
@@ -474,14 +474,53 @@ class CRegTestParams : public CChainParams {
     /**
      * Allows modifying the Version Bits regtest parameters.
      */
-    void UpdateVersionBitsParameters(Consensus::DeploymentPos d, int64_t nStartTime, int64_t nTimeout)
+    void UpdateVersionBitsParameters(Consensus::DeploymentPos d, int64_t startheight, int64_t timeoutheight)
     {
-        consensus.vDeployments[d].nStartTime = nStartTime;
-        consensus.vDeployments[d].nTimeout = nTimeout;
+        consensus.vDeployments[d].startheight = startheight;
+        consensus.vDeployments[d].timeoutheight = timeoutheight;
     }
+    bool CheckVBitsHeights(std::string& error, int64_t startheight, int64_t timeoutheight);
     void UpdateActivationParametersFromArgs(const ArgsManager& args);
 };
 
+bool CRegTestParams::CheckVBitsHeights(std::string& error, int64_t startheight, int64_t timeoutheight)
```

Comment:
In commit "Migrate versionbits to use height instead of MTP", but more general perhaps.

There seems to be some overlap here with `sanity_check_params`. Would it make sense to have them merged, and perhaps make it always performed (also for mainnet/testnet, at startup)?

Reply:
Seems best left for a refactor later IMO.

Reply:
I think it is reasonable to have the sanity check always performed at startup, but that can be left for a later PR.

However I have removed the overlap with `sanity_check_params` and have that instead call `CheckVBitsHeights` (now part of `CChainParams`) in the test for the height sanity checks.